lib/facebook_ads/model.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+require 'facebook_ads/json_converter'
+
+module FacebookAds
+  class Model
+    class << self
+      def field_converters
+        @field_converters ||= {}
+      end
+
+      def known_fields
+        field_converters.keys
+      end
+
+      def field(name, type: nil)
+        name = name.to_s
+        field_converters[name] = JsonConverter.for(type)
+
+        define_method name do
+          fields[name]
+        end
+
+        define_method "#{name}=" do |value|
+          fields[name] = value
+        end
+      end
+    end
+
+    attr_reader :id, :fields, :client
+
+    def initialize(id, client: FacebookAds.client)
+      @id = id
+      @fields = {}
+      @client = client
+    end
+
+    def fetch(fields)
+      fields = Array(fields)
+      validate_fields(fields)
+
+      response = client.get(id, params: { fields: fields.join(',') })
+      response.each do |field, value|
+        converter = self.class.field_converters[field]
+        next if converter.nil?
```

Comment:
Okay that explanation makes sense and I'm fine with simply skipping fields that were not requested.
I do agree that a missing field is an error. I'm unsure whether we should check for it just because it could be missing. If it happens and no error is raised it will be very annoying. On the other hand. I don't think that an API simply not providing a requested field (without raising an error) is rather unlikely and checking for it might a be a case of YAGNI. So... what do you think? Better safe than sorry?